python/createCollectionDownload.py

```diff
@@ -2,75 +2,189 @@
-    # ONCE YOU HAVE A URL FROM THE CHAINLISTDOWNLOADS ENDPOINT, THEN YOU'RE GOOD TO GO AND CAN DO WHATEVER YOU NEED WITH THAT FILE
-
-    fileGenerated = False
-    createdCollectionFileURL = False
+    print(f"Checking for status of downloaded file...")
     
-    while(fileGenerated == False):
-        print("Checking for status of generated file " + str(collection_download_id) + "...")
-        response = requests.get(url='https://location.chainxy.com/api/Downloads/{}'.format(collection_download_id), headers=headers)
-        r_body = json.loads(response.text)['Record']
+    while(file_generated == False):
+        
+        r = request_api(url=f'https://location.chainxy.com/api/Downloads/{collection_download_id}', headers=headers)
+        r_body = json.loads(r.text)['Record']
 
         if r_body['Status'] == 0:
-            print("Download " + str(collection_id) + " is still generating")
+            print(f"Download {str(collection_download_id)} is still generating...")
             time.sleep(5)
 
         elif r_body['Status'] == 2:
-            print("File generation failed. Speak to ChainXY for assistance")
-            fileGenerated = True
+            print("File generation failed. Please reach out to ChainXY support for assistance.")
+            file_generated = True
 
         elif r_body['Status'] == 1:
             print("File generation completed!")
-            fileGenerated = True
-            createdCollectionFileURL = r_body['Link']
+            file_generated = True
+            collection_download_link = r_body['Link']
+    
+    return collection_download_link
+
+def download_collection(cxy_api_key:str, collection_id:int, collection_type:str, cache_time:int = 24):
+    """
+    Downloads a collection based on the provided collection ID and collection type. An optional input cache_time determines if a new download should be made.
+    Params:
+        cxy_api_key:str - ChainXY API Key
+        collection_id:str - ID of the collection for which a new download will be initiated.
+        collectiontype:str - 'chain' or 'center' collection
+        cache_time:int - max. time in hours since the latest download before a new one is requested
+    Returns:
+        collection_download_link:str - link to S3 URL of the downloaded collection
+    """
+
+    # FOR MAKING REQUESTS TO THE API
+    headers = {
+        'x-apikey': cxy_api_key,
+        'x-Application': 'Python API Call',
+        'content-type': 'application/json'
+        }
+
+    validate_inputs(headers, cxy_api_key, collection_type, cache_time)    
+
+    records_url, check_url, new_download_url, url_params = get_collection_urls(collection_id, collection_type)
+    
+    records = json.loads(request_api(records_url, headers).text)['Records']
+
+    # IF A RECORD DOES NOT EXIST, CHECK IF THE COLLECTION EXISTS WITH 0 DOWNLOADS AND CREATE A NEW ONE;
+    # OTHERWISE, THERE IS NO DOWNLOAD TO BE MADE
+    if not records:
```

Comment:
this section needs to be revised, since this file doesn't deal with generating a new collection. This file is only meant for downloading existing collections. As such, functionality that deals with generating a new record etc. needs to be deleted here. You can use it to modify https://github.com/ChainXY/ChainXY_API/blob/main/python/createCollectionAndDownload.py that deals with creating a new collection.

Reply:
if you pursue this, I'd suggest splitting methods for generating ChainLists and SiteLists, the underlying entities are different enough that mixing them would just make the record creation very messy.

Reply:
Comment should be "If a download record doesn't exist, check if the collection exists with 0 downloads then create a new download." I'll change the comment